impl'd capability to retrieve console url (#400)

adietish opened the pull request:
fixes #400

adietish commented:
/ok-to-test

jeffmaury commented on src/main/java/com/openshift/internal/restclient/capability/server/Console.java:
```diff
@@ -0,0 +1,186 @@
+            String consoleResourceUrl = getOpenShiftResourceURL(resource, 
+                    Arrays.stream(OpenShiftConsoleResourceUrls.values())
+                            .filter(resourceUrl -> resourceUrl.getVersion() ==  3));
+            if (consoleResourceUrl != null) {
```
So if this is called for a resource type not supported by OpenShiftConsoleResourceUrls then null is returned but the builder has some kind of incomplete URL, is this on purpose ?

adietish replied:
if a resource type is not supported, then we should show the project overview. I added test (data points) that assert this.

jeffmaury commented on src/test/java/com/openshift/internal/restclient/capability/resources/BuildCapabilitiesIntegrationTest.java:
```diff
@@ -57,14 +57,14 @@ public void setUp() throws Exception {
         IBuildConfigBuilder builder = client.adapt(IBuildConfigBuilder.class);
         assertNotNull("Exp. the client to be able to use a buildconfigbuilder", builder);
-        LOG.debug("Creating BuildConfig {}", bc);
+        LOG.debug("Creating BUILDCONFIG {}", bc);
```
What's the rationale for that ?

adietish replied:
None, that's an unintentional change. Thanks for spotting. I'm correcting it.

adietish commented:
@jeffmaury 
> I would really like to see an integration test for this new capability
I added integration tests for the openshift major version and the console url. I added a simple test where I get the console url and http get it. Anything that goes beyond this is hard to test though:
* console root OS3: reports 403 for invalid URLs -> requesting and asserting 200 OK, works
* resources OS3: reports 200 for inexistant resource with an error in html as long as there's js enabled. If not it's complaining about missing js enablement -> dont see how I would test this 
* console root OS4: reports 200 for invalid URLs -> dont see how I would test this
* resources OS3: reports 200 for inexistant resource with an error in html as long as there's js enabled. If not it's complaining about missing js enablement -> dont see how I would test this

adietish commented:
/lgtm

adietish commented:
[merge]

adietish commented:
openshift-restclient-java 8.1.0-SNAPSTHO deployed to repository.jboss.org at https://repository.jboss.org/nexus/service/local/repositories/snapshots/content/com/openshift/openshift-restclient-java/8.1.0-SNAPSHOT/openshift-restclient-java-8.1.0-20191004.212928-1.jar
